### PR TITLE
Added support for multiple namespaces, timeout, error events and unit tests

### PR DESCRIPTION
**Pull request by rahulbhanushali**
The changes include:

- Support for multiple namespaces. A single node-app can specify multiple set of dependencies using namespaces and ready event will be fired for each set individually
- Timeout support - Error event will be emitted in case the dependencies are not resolved within the specified time
- Unit tests using tape
- Bumped version to 1.0

Breaking changes
- constructor and stun method now accepts objects as parameters

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,34 +1,100 @@
...
+        if (!options) {
+            options = {}
         }
-        if (!events && !eventsProvided) {
-            throw new Error('Initial events need to be present to stun')
+
+        var currentNS = options.namespace
+        var timeout = options.timeout
+
+        if (!currentNS) {
```
Comment (GavinDmello) [inline index.js]: short circuit ?

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,576 @@
+{
+  "name": "headrush",
```
Comment (GavinDmello) [inline package-lock.json]: Can you remove this file ?

File: index.js [hunk 3]
```diff
@@ -1,34 +1,100 @@
...
+            timeout = DEFAULT_TIMEOUT
+        }
+
+        NS[currentNS].timeout = timeout
+        NS[currentNS].context = this
+
+        var that = this
+        NS[currentNS].timeoutFunction = setTimeout(function () {
+            that.emit('error', {
+                code: 2,
```
Comment (GavinDmello) [inline index.js]: Why have error codes started from 2 ?

**Review by GavinDmello — CHANGES_REQUESTED** [review]

**Comment by GavinDmello** [comment]
@rahulbhanushali  Nice work, dude! Had no time to change this code.

File: index.js (hunk 1 above)
Comment (rahulbhanushali) [reply index.js]: We'll run into issues with short-circuit if someone chooses to use int values for namespaces like 0,1, and so on.

If the value passed in options = 0, the default values will be assigned always.

**Comment by rahulbhanushali** [comment]
@GavinDmello, have updated the PR.

**Pull request merged**